File: DataModule/src/com/mobnetic/coinguardian/model/CheckerInfo.java
package com.mobnetic.coinguardian.model;

import java.util.Locale;

public class CheckerInfo extends CurrencyPairInfo {

	public CheckerInfo(String currencyBase, String currencyCounter, String currencyPairId) {
		super(currencyBase, currencyCounter, currencyPairId);
	}
	
	public String getCurrencyBaseLowerCase() {
		return getCurrencyBase().toLowerCase(Locale.US);
	}
	
	public String getCurrencyCounterLowerCase() {
		return getCurrencyCounter().toLowerCase(Locale.US);
	}
}

File: dataModule/src/main/java/com/aneonex/bitcoinchecker/datamodule/model/CheckerInfo.kt
package com.aneonex.bitcoinchecker.datamodule.model

import java.util.*

class CheckerInfo(currencyBase: String, currencyCounter: String, currencyPairId: String?, contractType: FuturesContractType)
    : CurrencyPairInfo(currencyBase, currencyCounter, currencyPairId, contractType) {
    val currencyBaseLowerCase: String
        get() = currencyBase.lowercase(Locale.US)
    val currencyCounterLowerCase: String
        get() = currencyCounter.lowercase(Locale.US)
}